### ISSUE TITLE
animators

### ISSUE THREAD

**Issue by buhman** [buhman/hexagons#7]
Animators run independently of the event loop, and mutate models.

They have a duration and a start time. Their animation state is an interpolation of the current game time and start time + duration.

The event loop should provide this "real time" value.

---

I noticed many frameworks give a "delta-time from last loop iteration" value. This seems undesirable.

It would be easier to write functional animation code as an epoch-local delta, which would be interpolated with the animation function to produce the "current animation state" (IOW: does not depend on the object's last-interpolated state).

The event loop should then just provide a "real time" value, to make epoch-local deltas easier to calculate.

---

A token-path animator would have:

```
path (list of cubes)
epoch (ts)
duration (s)
```

The calculation would be something like:

```
delta-time = real-time - epoch
segment = (delta-time / duration) / path-segments
cube = interpolate(path[floor(segment)], path[ceil(segment)])
```

Instead of cube-nearest, rounding happens late, on the point rather than the cube (IOW: at any given animation state, the token could be (and probably is) on a fractional cube).

**Comment by buhman** [comment]
For tokens, this means the client event handler would emit an animator rather than mutate the token directly.

**Comment by buhman** [comment]
implemented as described in f29f4bd5b83f7285e5fb8de8ea3438ba0797d154 + c9a274595d597ea2b0d7e72caae34ed8600dcad1 + fab9106c0c550fac6cd38e92a8a2e9314e6cc809
